rtgui/locallabtools2.cc

```diff
@@ -4723,3 +4723,740 @@ void LocallabLog::updateLogGUI()
+        ToolParamBlock* const toolmaskBox = Gtk::manage(new ToolParamBlock());
+        
+        toolmaskBox->pack_start(*radmask, Gtk::PACK_SHRINK, 0);
+        toolmaskBox->pack_start(*lapmask, Gtk::PACK_SHRINK, 0);
+        toolmaskBox->pack_start(*chromask, Gtk::PACK_SHRINK, 0);
+        toolmaskBox->pack_start(*gammask, Gtk::PACK_SHRINK, 0);
+        toolmaskBox->pack_start(*slopmask, Gtk::PACK_SHRINK, 0);
+        toolmaskBox->pack_start(*shadmask, Gtk::PACK_SHRINK, 0);
+        toolmaskBox->pack_start(*mask_HCurveEditorG, Gtk::PACK_SHRINK, 4); // Padding is mandatory to correct behavior of curve editor
+        toolmaskBox->pack_start(*mask2CurveEditorG, Gtk::PACK_SHRINK, 4); // Padding is mandatory to correct behavior of curve editor
+        toolmaskBox->pack_start(*mask2CurveEditorGwav, Gtk::PACK_SHRINK, 4); // Padding is mandatory to correct behavior of curve editor
+        toolmaskBox->pack_start(*csThresholdmask, Gtk::PACK_SHRINK, 0);
+        toolmaskBox->pack_start(*gradFramemask, Gtk::PACK_SHRINK, 0);
+        toolmaskFrame->add(*toolmaskBox);
+        maskmaskBox->pack_start(*toolmaskFrame);
+        pack_start(*maskmaskBox);
+}
+
+bool LocallabMask::isMaskViewActive()
+{
+    return ((showmask_Method->get_active_row_number() != 0));
+}
+
+
+void LocallabMask::resetMaskView()
+{
+    showmask_MethodConn.block(true);
+
+    showmask_Method->set_active(0);
+
+    showmask_MethodConn.block(false);
+}
+
+void LocallabMask::getMaskView(int &colorMask, int &colorMaskinv, int &expMask, int &expMaskinv, int &shMask, int &shMaskinv, int &vibMask, int &softMask, int &blMask, int &tmMask, int &retiMask, int &sharMask, int &lcMask, int &cbMask, int &maskMask)
+{
+    maskMask = showmask_Method->get_active_row_number();
+}
+
+void LocallabMask::updateAdviceTooltips(const bool showTooltips)
+{
+    if (showTooltips) {
+        exp->set_tooltip_text(M("TP_LOCALLAB_MASKCOM_TOOLTIP"));
+        sensimask->set_tooltip_text(M("TP_LOCALLAB_SENSIMASK_TOOLTIP"));
+        blendmask->set_tooltip_text(M("TP_LOCALLAB_BLENDMASKMASK_TOOLTIP"));
+        CCmask_shape->setTooltip(M("TP_LOCALLAB_CURVEEDITORM_CC_TOOLTIP"));
+        LLmask_shape->setTooltip(M("TP_LOCALLAB_CURVEEDITORM_CC_TOOLTIP"));
+        HHmask_shape->setTooltip(M("TP_LOCALLAB_CURVEEDITORM_CC_TOOLTIP"));
+        struFrame->set_tooltip_text(M("TP_LOCALLAB_STRUMASK_TOOLTIP"));
+        mask_HCurveEditorG->set_tooltip_text(M("TP_LOCALLAB_HHMASK_TOOLTIP"));
+        radmask->set_tooltip_text(M("TP_LOCALLAB_LAPRAD_TOOLTIP"));
+        lapmask->set_tooltip_text(M("TP_LOCALLAB_LAPRAD_TOOLTIP"));
+        mask2CurveEditorG->set_tooltip_text(M("TP_LOCALLAB_WAVMASK_TOOLTIP"));
+        Lmask_shape->setTooltip(M("TP_LOCALLAB_LMASK_LL_TOOLTIP"));
+        mask2CurveEditorGwav->set_tooltip_text(M("TP_LOCALLAB_WAVMASK_TOOLTIP"));
+        LLmask_shapewav->setTooltip(M("TP_LOCALLAB_LMASK_LEVEL_TOOLTIP"));
+
+    } else {
+        exp->set_tooltip_text(M(""));
```

Comment:
@Desmis 
Should just write `exp->set_tooltip_text("")`: this avoid calling `M()` useless function here